pkcs12/tests/kdf.rs

```diff
@@ -150,14 +162,12 @@ fn pkcs12_key_derive_special_chars() {
     const PASS_SHORT: &str = "🔥";
     const SALT_INC: [u8; 8] = [0x1, 0x2, 0x3, 0x4, 0x5, 0x6, 0x7, 0x8];
 
-    assert_eq!(
-        derive_key_utf8::<sha2::Sha256>(
-            PASS_SHORT,
-            &SALT_INC,
-            Pkcs12KeyType::EncryptionKey,
-            100,
-            32
-        ),
-        hex!("d01e72a940b4b1a7a5707fc8264a60cb7606ff9051dedff90930687d2513c006")
-    );
+    assert!(derive_key_utf8::<sha2::Sha256>(
+        PASS_SHORT,
+        &SALT_INC,
+        Pkcs12KeyType::EncryptionKey,
+        100,
+        32
+    )
+    .is_err()); // Emoji is not in the Basic Multilingual Plane
```

Comment:
Note: now errors on the emoji test. I believe this is the correct behavior since the emoji is not valid BMP/UCS-2